dual_arm_teleop/src/robot_arm.cpp

```diff
@@ -0,0 +1,297 @@
+
+  // Initialize kinematic model
+  robot_model_loader::RobotModelLoader robot_model_load("robot_description");
+
+  kinematic_model = robot_model_load.getModel();
+  kinematic_state = std::make_shared<robot_state::RobotState>(kinematic_model);
+
+  joint_model_group = kinematic_model->getJointModelGroup(joint_model_group_name);
+
+  last_valid_pose = kinematic_state->getGlobalLinkTransform("victor_" + joint_model_group->getName() + "_link_7");
+  ee_start_translation = last_valid_pose.translation();
+  joint_position_measured.resize(7);
+
+  pub_arm = n.advertise<victor_hardware_interface::MotionCommand>(joint_model_group->getName() + "/motion_command", 10);
+  pub_gripper = n.advertise<victor_hardware_interface::Robotiq3FingerCommand>(joint_model_group->getName() + "/gripper_command", 10);
+  sub_arm_status = n.subscribe(joint_model_group->getName() + "/motion_status", 10, &RobotArm::updateMeasuredState, this);
+}
+
+void RobotArm::control(vive_msgs::ViveSystem msg)
+{
+  // If controller hand is invalid, stop
+  if (controller_hand != 1 && controller_hand != 2) return;
+
+  // Find the controller index who's hand we've been assigned
+  int assigned_controller_index = -1;
+  for (int i = 0; i < msg.controllers.size(); ++i)
+  {
+    if (msg.controllers[i].id == controller_hand)
+    {
+      assigned_controller_index = i;
+    }
+  }
+
+  // If there's no match, don't continue
+  if (assigned_controller_index == -1) return;
+
+  // A reference to the assigned controller
+  vive_msgs::Controller msg_controller = msg.controllers[assigned_controller_index];
+
+  // Update this arm's activation status
+  if (msg_controller.joystick.buttons[2] == 2) {
+    if (!trackpad_pressed) {
+      trackpad_pressed = true;
+      enabled = !enabled;
+    }
+  } else {
+    trackpad_pressed = false;
+  }
+
+  // Skip control if not enabled
+  if (!enabled) {
+    return;
+  }
+
+  // The controller pose in victor frame
+  //Eigen::Affine3d controller_pose = viveToVictorTranslation(pointMsgToEigen(msg_controller.posestamped.pose.position));
+
+  // Reset frame when button is pressed
+  if (msg_controller.joystick.buttons[0] == 2 || !initialized)
+  {
+    // Controller frame
+    controller_start_translation = pointMsgToEigen(msg_controller.posestamped.pose.position);
+
+    initialized = true;
+  }
+
+  // Reset orientation
+  if (msg_controller.joystick.buttons[1] == 2) {
+    controller_start_rotation = quatMsgToEigen(msg_controller.posestamped.pose.orientation);
+  }
+
+  // Calculate the relative pose
+  Eigen::Affine3d relative_pose = Eigen::Affine3d::Identity();
+
+  // Base
+  relative_pose.translate(ee_start_translation);
+
+  // Translation
+  Eigen::Vector3d translation(0, 0, 0);
+  translation += viveToVictorTranslation(pointMsgToEigen(msg_controller.posestamped.pose.position));
+  translation -= viveToVictorTranslation(controller_start_translation);
+  translation *= (msg_controller.joystick.axes[1] + 1.5) * .5; // motion scaling
```

Comment:
We probably dont want to hardcode these values.
We definitely do not want to hardcode them here.
At least put them at the top as a `#define`